compiler/src/dotty/tools/dotc/transform/localopt/Devalify.scala

```diff
@@ -194,7 +194,7 @@ class Devalify extends Optimisation {
       readingOnlyVals(qual)
 
     case t: Ident if !t.symbol.is(Mutable | Method) && !t.symbol.info.dealias.isInstanceOf[ExprType] =>
-      desugarIdent(t).forall(readingOnlyVals)
+      !isEffectivelyMutable(t)
```

Comment:
Did not understand why desugaring was needed here too

Reply:
I would keep the logic untouched then:

```scala
desugarIdent(t) match { case s: Select => readingOnlyVals(s); case _ => true }
```